### PR TITLE
code review - do not merge this

### PR DESCRIPTION
**Pull request by case-eee**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,45 @@
...
+Team members: Peter Manhart, George Lee, Nathan Werbeckes
+
+
+USER STORIES:
+
+as a user I want to be able to log in so I can interact with the website
+
+as a user I want to be able to see the website without logging in, but not be able to submit any votes or comments... so I can... see stuff
+
+as a user I want to be able to post a question so as to reach enlightenment
```
Comment (case-eee) [inline README.md:14]: :+1:

File: README.md [hunk 2]
```diff
@@ -1 +1,45 @@
...
+as a user I want to be able to up/down-vote questions, answers, and comments so other people can find "better" topics more easily
+
+as a user I want to only be able to vote on anything one time, so I can't cheat
+
+as a user I want to be able to see how many votes each question has on the main question index
+
+as a user I want to be able to edit my questions, answers, or comments so I can fix errors and/or clarify things
+
+
+STRETCH:
```
Comment (case-eee) [inline README.md:38]: :+1: on separating your user stories and prioritizing them based on importance.

File: app/controllers/login.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    else
+      redirect '/'
+    end
+  else
+    @error_messages = ["Invalid username and/or password"]
+    erb :'users/login'
+  end
+end
+
+get '/logout' do
```
Comment (case-eee) [inline app/controllers/login.rb]: since we are clearing the session here, the more RESTful convention is to make a `delete` request instead of a `get`

File: app/controllers/posts.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
+get '/' do
+  redirect '/index'
+end
+
+get '/index' do
+  @questions = Question.all
+  @answer_counts = @questions.map{ |question| question.answers.count }
+  @scores = @questions.map{ |question| question.score }
```
Comment (case-eee) [inline app/controllers/posts.rb:10]: the calculation for both `@scores` and `@answer_counts` should probably live on the `Question` model since it's performing a calculation on our data. when you have a chance to refactor (if you do), definitely extract this out!  skinny controllers - fat models (is more preferable than fat controllers)

File: app/controllers/posts.rb [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+get '/questions/new' do
+  puts "current user: #{current_user.username}"
+  erb :"posts/new"
+end
+
+post '/questions' do
+  puts params.inspect
+  puts "user id: #{session[:user_id]}"
+  puts "current user: #{current_user}"
+  @question = current_user.questions.create(params)
```
Comment (case-eee) [inline app/controllers/posts.rb:59]: :+1: on scoping the creation of a new question to the user. nice.

File: app/controllers/posts.rb [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  redirect '/index'
+end
+
+post '/questions/:id/answers' do
+  Answer.create(params[:answer])
```
Comment (case-eee) [inline app/controllers/posts.rb]: i highly suggest having some type of control here in case the answer doesn't save. that would be no bueno.

File: app/controllers/posts.rb [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  erb :"posts/question"
+end
+
+
+
+
+
+# this goes with the new-comment ajax request in application.js
+get '/comments/new' do
+  if request.xhr?
```
Comment (case-eee) [inline app/controllers/posts.rb:91]: woooo AJAX!!!

File: app/controllers/posts.rb [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    erb :'partials/_new_comment_form', layout: false
+  else
+    erb :'partials/_new_comment_form'
+  end
+end
+
+post '/comments' do
```
Comment (case-eee) [inline app/controllers/posts.rb:98]: where does the actual association of a comment to either a `Answer` or `Question` happen?

File: app/models/scoring.rb [hunk 9]
```diff
@@ -0,0 +1,10 @@
+module Scoring
```
Comment (case-eee) [inline app/models/scoring.rb:1]: nice :+1:

File: app/views/partials/_navbar.erb [hunk 10]
```diff
@@ -0,0 +1,24 @@
+<nav class="navbar navbar-inverse" role="navigation">
```
Comment (case-eee) [inline app/views/partials/_navbar.erb:1]: love that the navigation bar is a partial. this makes it easy to render on every page.

File: db/seeds.rb [hunk 11]
```diff
@@ -0,0 +1,31 @@
+require 'faker'
+
+User.create(username: "admin")
```
Comment (case-eee) [inline db/seeds.rb]: always use `create!` in your seed files (in case it doesn't actually get created, then you try to launch your app and have no data. just another thing to have to debug that we can easily avoid with `create!`)

File: app/views/layout.erb [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+  <link rel="stylesheet" href="/css/bootstrap.min.css">
+
+  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
+  <script src="/js/application.js"></script>
+
+  <title></title>
+</head>
+<body>
+  <%= erb :'partials/_navbar' %>
```
Comment (case-eee) [inline app/views/layout.erb:22]: :tada: very niceee

File: app/models/question.rb [hunk 13]
```diff
@@ -0,0 +1,16 @@
+class Question < ActiveRecord::Base
+  include Scoring
+  validates :content, presence: true
+  validates :title, presence: true
+
+  belongs_to :user
```
Comment (case-eee) [inline app/models/question.rb:6]: like i mentioned briefly yesterday.. if you have a `belongs_to` that is required - you probably want to add a validation on that foreign_key field to ensure that it is associated with a parent object before it's saved. we don't want bad data in our database

File: public/js/bootstrap.js [hunk 14]
```diff
@@ -0,0 +1,2317 @@
+/*!
+ * Bootstrap v3.3.4 (http://getbootstrap.com)
```
Comment (case-eee) [inline public/js/bootstrap.js:2]: bootstrap!!

File: public/js/application.js [hunk 15]
```diff
@@ -0,0 +1,71 @@
+$(document).ready(function() {
+  // This is called after the document has loaded in its entirety
+  // This guarantees that any elements we bind to will exist on the page
+  // when we try to bind to them
+
+  // See: http://docs.jquery.com/Tutorials:Introducing_$(document).ready()
+
+
+  // trying to add new-comment button functionality here:
+  $('.comment_link').click(function(event){
```
Comment (case-eee) [inline public/js/application.js]: nice work so far on making AJAX requests. keep AJAXing to enhance a user's experience while interacting with your app! 

these AJAX requests looks solid to me. continue practicing.

**Comment by case-eee** [comment]
@petermanhart @ghrlee @werbeckes good work so far - overall, this looks awesome. i just pointed out things that you could improve. you have plenty of time today to finish this app. 

btw - this is typically a project at the end of week one in phase three. and y'all are killing it right now! keep it up :tada: :tada:

File: app/controllers/posts.rb (hunk 4 above)
Comment (petermanhart) [inline app/controllers/posts.rb]: note to team - this is done

File: db/seeds.rb (hunk 11 above)
Comment (petermanhart) [inline db/seeds.rb]: this is also done

File: app/models/question.rb (hunk 13 above)
Comment (petermanhart) [inline app/models/question.rb:6]: done
